bundles/documents-rest-api/src/FondOfOryx/Glue/DocumentsRestApi/Processor/Reader/DocumentReader.php

```diff
@@ -0,0 +1,119 @@
+
+    /**
+     * @var array<\FondOfOryx\Glue\DocumentsRestApi\Dependency\Plugin\DocumentTypePluginInterface>
+     */
+    protected array $documentTypePlugins;
+
+    /**
+     * @param \FondOfOryx\Glue\DocumentsRestApi\Dependency\Client\DocumentsRestApiToEasyApiInterface $client
+     * @param \FondOfOryx\Glue\DocumentsRestApi\Processor\Mapper\DocumentRestRequestMapperInterface $requestMapper
+     * @param \FondOfOryx\Glue\DocumentsRestApi\Processor\Builder\RestResponseBuilderInterface $responseBuilder
+     * @param \Psr\Log\LoggerInterface $logger
+     * @param array<\FondOfOryx\Glue\DocumentsRestApi\Dependency\Plugin\DocumentTypePluginInterface> $documentTypePlugins
+     */
+    public function __construct(
+        DocumentsRestApiToEasyApiInterface $client,
+        DocumentRestRequestMapperInterface $requestMapper,
+        RestResponseBuilderInterface $responseBuilder,
+        LoggerInterface $logger,
+        array $documentTypePlugins
+    ) {
+        $this->client = $client;
+        $this->requestMapper = $requestMapper;
+        $this->responseBuilder = $responseBuilder;
+        $this->logger = $logger;
+        $this->documentTypePlugins = $documentTypePlugins;
+    }
+
+    /**
+     * @param \Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface $restRequest
+     *
+     * @return \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResponseInterface
+     */
+    public function find(RestRequestInterface $restRequest): RestResponseInterface
+    {
+        $filter = $this->requestMapper->fromRestRequest($restRequest);
+        $easyApiFilter = null;
+        foreach ($this->documentTypePlugins as $documentTypePlugin) {
+            if ($documentTypePlugin->getType() === $filter->getType()) {
+                $easyApiFilter = $documentTypePlugin->createEasyApiFilter($filter);
+            }
+        }
+
+        if ($easyApiFilter === null) {
+            $this->logger->error(sprintf('Missing plugin for document type "%s"', $filter->getType()));
+
+            return $this->responseBuilder->buildErrorRestResponse();
+        }
+
+        $error = $easyApiFilter->getError();
+        if ($error !== null) {
+            $this->logger->error(sprintf('%s: %s | %s', $filter->getId(), $error->getMessage(), $error->getValue()));
+
+            return $this->responseBuilder->buildErrorRestResponse($error->getMessage(), $error->getValue(), (int)$error->getValue());
+        }
+
+        $response = $this->client->findDocument($easyApiFilter);
+
+        if ($response->getStatus() !== 'success' || $response->getHash() !== sha1($response->getData())) {
+            return $this->responseBuilder->buildErrorRestResponse();
+        }
+
+        try {
+            $data = json_decode($response->getData(), true);
+
+            $document = (new DocumentTransfer())->fromArray($data, true);
+
+            $test = (new EasyApiRequestTransfer())->setUri($this->resolveAttachmentUri($document));
```

Comment:
@julianzimmermann $test? Is there a reason, why you the paramater is named as $test

Reply:
nope, thx